### ISSUE TITLE
torch.norm produces wrong results 

### ISSUE THREAD

**Issue by XiaoXiao-Woo** [pytorch/pytorch#140282]
### 🐛 Describe the bug

```python
import torch
import numpy as np

TV = torch.load("/home/dsq/NIPS/opt/siren/experiment_scripts/TV_e.pth").cpu()

print(TV.shape)

print(torch.linalg.norm(TV).item())

print(torch.linalg.norm(torch.linalg.norm(TV, dim=-1, keepdim=True)).item())

print(np.linalg.norm(TV.cpu().detach().numpy()).item())

print(torch.linalg.norm(torch.sqrt(TV[:, 0] ** 2 + TV[:, 1] ** 2)).item())
```

2.5.1+cu12.4, numpy 1.26.4 results: 
torch.Size([38000, 2])
141.7647247314453
141.76470947265625
141.7647705078125
141.76475524902344

2.4.1+cu12.4, numpy 1.24.4 results:
torch.Size([38000, 2])
141.7647247314453
141.76470947265625
141.76475524902344
141.76475524902344

Expected results:
141.76475524902344

### Versions

2.5.1+cu12.4, 2.4.1+cu12.4, and 1.13.1+cu11.7

**Comment by malfet** [comment]
See https://pytorch.org/docs/main/notes/numerical_accuracy.html
Can you please clarify why do you think results should be one vs another and why error exceeds numerical accuracy expectations for a datatype
